### PR TITLE
hack the tower feedback updates

### PR DESCRIPTION
**Pull request by nhojpatrick**
- update to newer ubuntu lts 14.04 instead of 12.04
- correct where hg clone to match the other documentation
- fix date time compile error on 1st boot by force date time sync pre hg clone
- add scripts for jigsaw

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -8,9 +8,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
 	config.vm.hostname = "adoptopenjdk"
 
-	config.vm.box = "ubuntu-#{ENV['VAGRANT_BOX_NAME_SUFFIX']}/precise-amd64"
+	config.vm.box = "ubuntu/trusty-amd64"
```
Comment (neomatrix369) [inline Vagrantfile]: This line is missing the env variable```#{ENV['VAGRANT_BOX_NAME_SUFFIX']}``` - do you have a reason not to include it

File: Vagrantfile [hunk 2]
```diff
@@ -8,9 +8,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
 
-	config.vm.box_url = "https://cloud-images.ubuntu.com/vagrant/precise/current/precise-server-cloudimg-amd64-vagrant-disk1.box"
+	config.vm.box_url = "https://cloud-images.ubuntu.com/vagrant/trusty/current/trusty-server-cloudimg-amd64-vagrant-disk1.box"
+
+	config.vm.provision :shell, :inline => "sudo /vagrant/scripts/datetime-fix.sh"
```
Comment (neomatrix369) [inline Vagrantfile]: good one :+1:

File: buildJigsawOpenJDK9UsingVagrant.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export VAGRANT_BOX_CPUS=2
```
Comment (neomatrix369) [inline buildJigsawOpenJDK9UsingVagrant.sh]: :+1:

File: scripts/setupEnvForOpenJDK9.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export BUILD_NAME="OpenJDK9"
 export JDK_FOLDER=jdk9
-export OPENJDK_REPO=http://hg.openjdk.java.net/jdk9/dev
+export OPENJDK_REPO=http://hg.openjdk.java.net/jdk9/jdk9
```
Comment (neomatrix369) [inline scripts/setupEnvForOpenJDK9.sh]: :+1:

**Comment by neomatrix369** [comment]
All changes look good, apart from my query.

File: Vagrantfile (hunk 1 above)
Comment (nhojpatrick) [inline Vagrantfile]: #{ENV['VAGRANT_BOX_NAME_SUFFIX']} is not required, as I'm unsure why it was being used previously.

the vagrant box we are using is a standard ubuntu image, it has nothing to do with OpenJDK so it seams wrong to put OpenJDK into the box name. Theoretically other projects could use the exact same box.

Think of `config.vm.box` as the name of based vm to use. When you execute `bash vagrant up` if no vm has been created, vagrant checks that name against `vagrant box list` and if it doesn't exist then it downloads it from `config.vm.box_url`. Once download it registers that downloaded box against the name `config.vm.box`. Vagrant then clones the box for your project, then boots the box as a vm. Once cloned that box is OpenJDK specific and is given a generated name to appear in Virtual Box.

Hope that helps you understand why I believe the env argument is not required in the `config.vm.box` name.

**Comment by nhojpatrick** [comment]
answers inline above regarding why I believe `#{ENV['VAGRANT_BOX_NAME_SUFFIX']}` is not required.

File: Vagrantfile (hunk 1 above)
Comment (neomatrix369) [inline Vagrantfile]: Thanks for the explanation.

**Comment by neomatrix369** [comment]
Good to merge.

**Pull request merged**